### ISSUE TITLE
Containers should be empty when no options are provide and it doesn't have a default

### ISSUE THREAD

**Issue by inkychris** [henryiii/CLI11#886]
I would expect the resulting vector of values returned by CLI11 in the following example to be empty when the option isn't used on the command line, but instead it contains a single default-initialised element instead. The work around is to manually check if the option was used via `count()` before attempting to iterate over the resulting container. It would be nice if we could rely on an empty container returned by CLI11 to indicate that no options were provided. When the option has a default value set and no options provided, the vector contains the default value as expected.

```c++
#include <vector>

#include "CLI/CLI.hpp"

int main(int argc, char** argv) {
    CLI::App app { "test" };
    app.add_option("--value")
        ->expected(0, -1);

    CLI11_PARSE(app, argc, argv);

    auto values = app.get_option("--value")->as<std::vector<int>>();

    for (auto value : values) {
        std::cout << "value: " << value << "\n";
    }

    return 0;
}
```
with no arguments, prints:
```
value: 0
```

**Comment by phlptp** [comment]
There is a bit of a philosophical debate here.  The way CLI treats it by default is as a vector of arguments, each can be empty or have a value.   It admittedly gets a little fuzzy when requesting the value of an argument of which nothing was given,  but in that case it was decided the first argument of a returned vector should be the same as if you requested as single string, which means an empty string or whatever an empty string converts to in the type requested.  

The way around this is to set the default value to the string "{}"  this indicates that an empty vector/container is to be returned if no arguments are given.

**Comment by inkychris** [comment]
Ah thanks for the clarification! I was going to offer to update the docs but it looks like the book has already been updated in source to talk about empty containers, it just doesn't seem to have been updated on the docs site?

**Comment by phlptp** [comment]
Will close this when I figure out what is going with the docs and how to update that.  Thanks,

**Comment by henryiii** [comment]
Fixed in #897 & now live.
